src/Amalgam/Parser.cpp

```diff
@@ -450,18 +450,20 @@ EvaluableNode *Parser::GetNextToken(EvaluableNode *new_token)
 			return nullptr;
 		}
 
-		std::string token = GetNextIdentifier();
+		std::string token_string = GetNextIdentifier();
+		EvaluableNodeType token_type = GetEvaluableNodeTypeFromString(token_string);
 		//first see if it's a keyword
-		new_token->SetType(GetEvaluableNodeTypeFromString(token), evaluableNodeManager, false);
-		if(IsEvaluableNodeTypeValid(new_token->GetType()))
+		new_token->SetType(token_type, evaluableNodeManager, false);
+
+		if(IsEvaluableNodeTypeValid(token_type) && !IsEvaluableNodeTypeImmediate(token_type))
```

Comment:
should this have `IsEvaluableNodeTypeValid(token_type->GetType())` or was the old code incorrect?